lib/client/client.js

```diff
@@ -160,10 +160,33 @@ AccountsTemplates.submitCallback = function(error, state, onSuccess) {
 
   if (error) {
     if (_.isObject(error.details)) {
-      // If error.details is an object, we may try to set fields errors from it
-      _.each(error.details, function(error, fieldId) {
-        AccountsTemplates.getField(fieldId).setError(error);
-      });
+      if (error.error === 'validation-error') {
+        // This error is a ValidationError from the mdg:validation-error package.
+        // It has a well-defined error format
+
+        // Record errors that don't correspond to fields in the form
+        var errorsWithoutField = [];
+
+        _.each(error.details, function(fieldError) {
+          var field = AccountsTemplates.getField(fieldError.name);
+
+          if (field) {
+            // XXX in the future, this should have a way to do i18n
+            field.setError(fieldError.type);
+          } else {
+            errorsWithoutField(fieldError.type);
```

Comment:
Oh, you're right. I guess there aren't really any tests, maybe I should add some.